src/main/java/com/openshift/internal/restclient/okhttp/AuthenticatorInterceptor.java

```diff
@@ -84,21 +79,21 @@ private boolean isUrlWithoutAuthorization(Request request, String url) {
         return url.endsWith(DefaultClient.PATH_OPENSHIFT_VERSION)
                 || url.endsWith(DefaultClient.PATH_KUBERNETES_VERSION)
                 || url.endsWith(DefaultClient.PATH_OAUTH_AUTHORIZATION_SERVER)
-                || request.url().toString().startsWith(client.getAuthorizationEndpoint().toString());
+                || url.startsWith(client.getAuthorizationEndpoint().toString());
     }
 
     private Response authenticate() throws IOException {
         OkHttpClient okClient = client.adapt(OkHttpClient.class);
         if (okClient == null) {
             return null;
         }
-        Request authRequest = new Request.Builder()
-                .addHeader(CSRF_TOKEN, "1")
-                .url(new URL(client.getAuthorizationEndpoint().toExternalForm() 
-                        + "?response_type=token&client_id=openshift-challenging-client").toString())
-                .build();
+        Request authRequest = appendAuthorization(
+                client.getAuthorizationContext(),
+                new Request.Builder()
+                    .addHeader(CSRF_TOKEN, "1")
+                    .url(new URL(client.getAuthorizationEndpoint().toExternalForm() 
+                            + "?response_type=token&client_id=openshift-challenging-client").toString()));
```

Comment:
explanation: request the bearer token without authenticator which is an unnecessary indirection)